### PR TITLE
Proposal for rule 303

### PR DESCRIPTION
**Pull request by rknegjens**
> **303.** All player have a "voting karma" (VK) score, which is separate to the points defined in rule 112. All players begin with zero VK. A player's voting karma is modified based on how promptly they perform the voting actions of *proposing* a rule and *voting* on a rule when required to. If a player does either action *on time* (defined below) their VK score is incremented by one. If they are late to perform either action their VK score is decremented by the number of days they exceeded the *on time* deadline, rounded to the nearest 24 hours.

> In order to *propose* a rule *on time* the proposal must be submitted no later than 48 hours after midnight (UTC) of the day that the previous rule was voted on.
In order to *vote* on a rule *on time* the vote must be cast no later than 24 hours after midnight (UTC) of the day the proposal was *submitted* or last *amended* or last *discussed*.
A rule is *amended* when the proposer updates the rule via a new git commit. A rule is *discussed* when a player comments on the pull request of the rule. Successive comments by the same player without a response from other players are ignored i.e. only the time of the first comment should be considered in this case.

This rule introduces a karma point system. In a future rule we can decide how/if to couple this to the real points e.g. when people reach positive or negative karma thresholds.

### CODE REVIEW

**Comment by roarandrock** [comment]
This sounds like potentially a lot of upkeep work. But I'm in favor of faster turns. So I approve.

**Review by roarandrock — APPROVED** [review]

**Comment by rknegjens** [comment]
There is an edge case regarding voting with github: a merge review can be approved by one player and then later modified by a new commit. Should this reset the vote? For the sake of rule 303 I think not i.e. your T is your first vote. However for the game itself it would seem fair to require a re-vote.

**Comment by RamonRB** [comment]
What would the VK be used for? Currently the rule does not specifiy.

Edit: apologies, you specified in a rule clarification. However, I'm still not entirely sure what the idea is/ Would it be used for tie breaking or such? Otherwise, it might be a lot of work for no benefit.

**Comment by rknegjens** [comment]
> What would the VK be used for? Currently the rule does not specifiy.

As I wrote in extra explanation above that's for a future rule to decide. For now they're just kudos points with no effect on the actual score. If we want them to be more effective in future we could for example couple them directly to real points, set a threshold (-5, or +5) at which point you gain/lose real points etc.

**Comment by akvadrako** [comment]
I don’t know about this in general. But one issue is that it encourages people to vote right away, but maybe they want to discuss changes first. Though I suppose since the karma points are separate, the only real downside is the added complexity.

I will think about it more.

**Comment by rknegjens** [comment]
@akvadrako 
To promote more discussion the T for voting can be reset after every update (commit or comment, not a vote) on the proposed rule's pull request. Successive updates from same player are counted from the first. The X for voting could then be lowered to 2 (or 1?).

A discussion can then go on indefinitely if there is a regular back and forth between players. Once this stops, everyone must vote within X.

**Comment by rknegjens** [comment]
@RamonRB VK encourages timely rule proposals and voting by keeping track of player's punctuality. Players can take pride in a high kudos score, or be ashamed of a negative one etc. and it will also be objectively visible who the culprits are.

It's up to a future rule to decide how to link it to the real points if desirable. Anything is possible, it could be a tiebreaker, but more effective could be a direct boost to a player's score e.g. 1 VK could equal 5 points, or crossing a the +/- 5 VK threshold could add/deduct 15 points, etc. The idea is that by not being punctual you hurt your chances of winning by lowering your score, so you're encouraged to be timely both fir the good of the game and your chances of winning.

**Comment by RamonRB** [comment]
Since I'm not keeping score I can easily approve this, so I leave it to the bookkeeper to veto this or not. I'm not conviced on the usefulness, but have no objections either. And we can always repeal it if it turns out to be too much hassle.

**Review by RamonRB — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
------------ | ----- | ------
-akvadrako   | Ba    | -2
-RamonRB     | Be    | 11
-rknegjens   | K     | 0
-roarandrock | W     | 0
+player      | order | points | VK
+----------- | ----- | ------ | -- 
+akvadrako   | Ba    | -2     | 0
+RamonRB     | Be    | 11     | 0
+rknegjens   | K     | 0      | 0
```
Comment (akvadrako) [inline README.md]: You could also adjust all the points and VK levels here, like if the PR is accepted.

**Comment by akvadrako** [comment]
I find the explanation kinda hard to understand. So, it's:

- All players earn 1 VK every turn.
- The deadline for proposals is 48 hours from the turn start.
- The deadline for voting is 72 hours from the proposal. 
- Players who propose or vote late lose VK, one per 24 hours (rounded up).

Also I think you need to add that a turn starts after the last turn ends. And when does the last turn end? All the things from my PR. I think midnight UTC is the best.

**Comment by rknegjens** [comment]
@akvadrako 
Did you see my last messaged directed to you? Here I propose some changes to stimulate more discussion (which you missed in your summary). If you agree I'll incorporate it into my rule and clarify the points you mentioned.

Note that all players can _potentially_ earn 1 VK per turn. The timer (T) starts once the last turn is "commited" in the case of the proposal phase, or the last "commit/comment" on the PR in the voting phase.

**Comment by akvadrako** [comment]
Yes I saw that, I didn't mean to make the summary complete. I was trying to simplify the logic so it doesn't require formulas :) I thought it's simpler to say they earn VK then lose VK for late actions, but one could also say they can earn VK if on time or lose VK if actions are over a day late.

I don't like the last commit being used as the start time because it requires keeping track of what time the last commit happened and always referring to that when doing VK scoring. That's why I suggest to start new turns at midnight. In either case it needs to be part of the rule.

**Comment by rknegjens** [comment]
@akvadrako @roarandrock @RamonRB @fpug 
I've updated the rule based on some good input from Devin. Please check and re-vote if necessary.

**Review by akvadrako — APPROVED** [review]

**Review by fpug — APPROVED** [review]

**Pull request merged**